### PR TITLE
Devops/split logs

### PR DESCRIPTION
**Pull request by Hikaru2035**
# Ticket
#668

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [x] PR has at least 1 peer approval before senior review.
- [x] I fully understand the code I submitted.
- [x] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: indexer/internal/log/logger.go [hunk 1]
```diff
@@ -67,16 +68,29 @@ func NewLogger(name string) zerolog.Logger {
...
 func createFileWriter(logConfig config.LogConfig) (io.Writer, error) {
-	// For now, we'll use a simple file writer
-	// In production, you might want to use lumberjack for rotation
-	file, err := os.OpenFile(logConfig.FilePath, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0o666)
-	if err != nil {
-		return nil, err
+	// Set default values for rotation if not specified
+	maxSize := logConfig.MaxSize
+	if maxSize == 0 {
+		maxSize = 100 // 100MB default
```
Comment (vietdangquoc) [inline indexer/internal/log/logger.go]: pls using const

File: indexer/internal/log/logger.go [hunk 2]
```diff
@@ -67,16 +68,29 @@ func NewLogger(name string) zerolog.Logger {
...
 	}
 
-	// Note: For production use, consider implementing log rotation
-	// using a library like gopkg.in/natefinch/lumberjack.v2
-	return file, nil
+	maxBackups := logConfig.MaxBackups
+	if maxBackups == 0 {
+		maxBackups = 10 // Keep max 10 old log files
```
Comment (vietdangquoc) [inline indexer/internal/log/logger.go]: pls using const

File: indexer/internal/log/logger.go [hunk 3]
```diff
@@ -67,16 +68,29 @@ func NewLogger(name string) zerolog.Logger {
...
+	}
+
+	maxAge := logConfig.MaxAge
+	if maxAge == 0 {
+		maxAge = 30 // 30 days default
```
Comment (vietdangquoc) [inline indexer/internal/log/logger.go]: pls using const

**Review by vietdangquoc — APPROVED** [review]

**Comment by sontung-ncc** [comment]
change PR to develop

**Comment by Hikaru2035** [comment]
cherry-pick to merge to develop: https://github.com/mezonai/mmn-tx-explorer/pull/910
